src/arch/x86_64/mod.rs

```diff
@@ -319,6 +319,15 @@ mod tests {
 		assert!(freq < 10000); //More than 10Ghz is probably wrong
 	}
 
+	#[test]
+	fn detect_freq_from_sysinfo() {
+		let freq_res = crate::detect_freq_from_sysinfo();
+		assert!(freq_res.is_ok());
+		let freq = freq_res.unwrap();
+		assert!(freq > 0);
+		assert!(freq < 10000);
```

Comment:
```suggestion
		assert!(freq > 50000 && freq < 10000000);
```
- I've never seen a system with less than 100kHz that would be capable of running uhyve
- We are talking about kHz here, right?

Reply:
If I run `rusty_demo` on my machine, I get the following output:

```
[0][INFO] ========================== CPU INFORMATION ===========================
[0][INFO] Model:                   uhyve - unikernel hypervisor
[0][INFO] Frequency:               3267 MHz (from Hypervisor)
```

> We are talking about kHz here, right?

The answer is **no**. Other tests like `detect_freq_from_cpuid_hypervisor_info` and `get_cpu_frequency_from_os` get the value in Hz and then convert it to MHz using the `MHZ_TO_HZ` constant defined at the beginning of the file. Unless if I am extremely confused, such a conversion should not be necessary here - but I will make sure to add a comment to make it clearer.

Like `lscpu` and `cat /proc/cpuinfo`, the value returned is in `MHz`, despite the `sysinfo` crate not mentioning that explicitly.

Reply:
If I run `rusty_demo` on my machine, I get the following output:

```
[0][INFO] ========================== CPU INFORMATION ===========================
[0][INFO] Model:                   uhyve - unikernel hypervisor
[0][INFO] Frequency:               3267 MHz (from Hypervisor)
```

> We are talking about kHz here, right?

The answer is **no**. Other tests like `detect_freq_from_cpuid_hypervisor_info` and `get_cpu_frequency_from_os` get the value in Hz and then convert it to MHz using the `MHZ_TO_HZ` constant defined at the beginning of the file. Unless if I am extremely confused, such a conversion should not be necessary here - but I will make sure to add a comment to make it clearer.

Like `lscpu` and `cat /proc/cpuinfo`, the value returned is in `MHz`, despite the `sysinfo` crate not mentioning that explicitly.